client/signup/steps/user/index.jsx

```diff
@@ -488,7 +488,11 @@ export class UserStep extends Component {
 	}
 
 	renderSignupForm() {
-		const { oauth2Client, isReskinned, isPasswordless } = this.props;
+		const { oauth2Client, isReskinned } = this.props;
+		const isPasswordless =
+			isMobile() ||
+			this.props.isPasswordless ||
+			isNewsletterFlow( this.props?.queryObject?.variationName );
```

Comment:
I was concerned that there is flow-specific code here inside the step while there also is a prop controlled externally. My gut reaction was that all these conditions should be external.

That said, external prop is passed from the config — example for hosting flow:

https://github.com/Automattic/wp-calypso/blob/d68ac9d00997aafb2aec0042d60dd6d6006885e8/client/signup/config/steps-pure.js#L194-L197

Newsletter flow doesn't have Start framework config, but would be still better to be able to pass it externally. I can't think of convenient way to do that, though.

Until then, this seems fine.